### ISSUE TITLE
`docker-compose ps` errors while talking with docker

### ISSUE THREAD

**Issue by ff-** [docker/compose#2697]
I get the following error while trying to do `docker-compose ps` or `docker-compose pull`. Happens with 1.6.0-RC1 and 1.5.2 (well in reality 1.5.2 blows up only when installing from pip, so I guess the bug is in `docker-py`).
Docker version: 1.9.1

```
# docker-compose pull
Traceback (most recent call last):
  File "<string>", line 3, in <module>
  File "/code/compose/cli/main.py", line 54, in main
  File "/code/compose/cli/docopt_command.py", line 23, in sys_dispatch
  File "/code/compose/cli/docopt_command.py", line 26, in dispatch
  File "/code/compose/cli/main.py", line 171, in perform_command
  File "/code/compose/cli/command.py", line 52, in project_from_options
  File "/code/compose/cli/command.py", line 85, in get_project
  File "/code/compose/cli/command.py", line 66, in get_client
  File "/code/compose/cli/docker_client.py", line 27, in docker_client
  File "/code/.tox/py27/lib/python2.7/site-packages/docker/client.py", line 56, in __init__
  File "/code/.tox/py27/lib/python2.7/site-packages/docker/auth/auth.py", line 177, in load_config
  File "/code/.tox/py27/lib/python2.7/site-packages/docker/auth/auth.py", line 118, in parse_auth
TypeError: string indices must be integers
docker-compose returned -1
```

Of course the docker daemon is up and in good health.

**Comment by ff-** [comment]
Installing with pip seems to give a more useful traceback:

```
Traceback (most recent call last):
  File "/usr/local/bin/docker-compose", line 9, in <module>
    load_entry_point('docker-compose==1.5.2', 'console_scripts', 'docker-compose')()
  File "/usr/local/lib/python2.7/dist-packages/compose/cli/main.py", line 54, in main
    command.sys_dispatch()
  File "/usr/local/lib/python2.7/dist-packages/compose/cli/docopt_command.py", line 23, in sys_dispatch
    self.dispatch(sys.argv[1:], None)
  File "/usr/local/lib/python2.7/dist-packages/compose/cli/docopt_command.py", line 26, in dispatch
    self.perform_command(*self.parse(argv, global_options))
  File "/usr/local/lib/python2.7/dist-packages/compose/cli/main.py", line 169, in perform_command
    project = project_from_options(self.base_dir, options)
  File "/usr/local/lib/python2.7/dist-packages/compose/cli/command.py", line 53, in project_from_options
    network_driver=options.get('--x-network-driver'),
  File "/usr/local/lib/python2.7/dist-packages/compose/cli/command.py", line 89, in get_project
    get_client(verbose=verbose, version=api_version),
  File "/usr/local/lib/python2.7/dist-packages/compose/cli/command.py", line 70, in get_client
    client = docker_client(version=version)
  File "/usr/local/lib/python2.7/dist-packages/compose/cli/docker_client.py", line 28, in docker_client
    return Client(**kwargs)
  File "/usr/local/lib/python2.7/dist-packages/docker/client.py", line 56, in __init__
    self._auth_configs = auth.load_config()
  File "/usr/local/lib/python2.7/dist-packages/docker/auth/auth.py", line 175, in load_config
    return parse_auth(json.load(f))
  File "/usr/local/lib/python2.7/dist-packages/docker/auth/auth.py", line 116, in parse_auth
    username, password = decode_auth(entry['auth'])
TypeError: string indices must be integers
```

**Comment by dnephin** [comment]
This looks like a bug in `docker-py` when reading your docker config. Could you include a sanitized version of your docker config file?

**Comment by ff-** [comment]
If you mean `/etc/default/docker`, everything's commented there.
Instead `/etc/docker/key.json` contains a k:v dictionary for the elliptic curve configuration, where keys and values are strings.

**Comment by dnephin** [comment]
Sorry, I meant the user config for the client. It should be in either `~/.docker/config.json` or (the legacy path) `~/.dockercfg` (unless you set `$DOCKER_CONFIG` to something else).

Either way, this is an issue with docker-py. Could you open an issue on https://github.com/docker/docker-py and include this traceback, as well as the contents of the client config file, and the docker-py version (which you can get from either `docker-compose version` or `pip freeze | grep docker`).

Looking at the code, I think it might be an issue handling the legacy config format.

**Comment by shin-** [comment]
This would be fixed by #2992
